proxy/request_test.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import (
+	"testing"
+)
+
+func TestBadMessage(t *testing.T) {
+	req := ""
+	resp := handleRequest([]byte(req))
+
+	expected := `{"id":null,"error":{"errcode":"M_NOT_JSON","error":"unexpected end of JSON input"}}`
+	if string(resp) != expected {
```

Comment:
Asserting on stringification of JSON which is non-deterministically specified seems poor form.

Make a jsonResponse object, and compare that they're equal, instead?